internal/services/compute/virtual_machine_scale_set.go

```diff
@@ -1813,6 +1813,10 @@ func VirtualMachineScaleSetRollingUpgradePolicySchema() *pluginsdk.Schema {
 					Type:     pluginsdk.TypeBool,
 					Optional: true,
 				},
+				"max_surge": {
```

Comment:
my only comment might be to be a tad more clear on what this is doing such as
```suggestion
				"maximum_surge_instances": {
```